src/python/ensembl/genes/tracking/bioproject_tracking.py

```diff
@@ -0,0 +1,250 @@
+        >>> rank = 'order'
+        >>> updated_annotations = get_taxonomy_info(live_annotations, bioproject_accessions_taxon, rank)
+        >>> print(updated_annotations)
+        {'accession1': {'some_annotation': 'value', 'order': 'SomeOrderName'}}
+    """
+    for accession in live_annotations:
+        taxon_id = bioproject_accessions_taxon[accession]["taxon_id"]
+        url = f"https://api.ncbi.nlm.nih.gov/datasets/v2alpha/taxonomy/taxon/{taxon_id}/dataset_report"
+        
+        try:
+            response = requests.get(url)
+            response.raise_for_status()  # Raises an HTTPError if the response status code is 4XX or 5XX
+            data = response.json()
+            taxonomies = data.get('reports', [])
+            for taxonomy in taxonomies:
+                rank_name = taxonomy.get('taxonomy', {}).get('classification', {}).get(rank, {}).get('name')
+                taxonomy_info = {rank : rank_name}
+                live_annotations[accession].update(taxonomy_info)
+                
+        except requests.HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"An error occurred: {err}")
+           
+    return(live_annotations)
+        
+def write_report(live_annotations, report_file, rank):
+    with open(Path(report_file), "w", encoding="utf8") as report_file:
+        for accession in live_annotations:
+            try:
+                report_file.write(accession + '\t' + live_annotations[accession]["dbname"] + '\t' + live_annotations[accession][rank] + '\n')
+            except TypeError:
+                report_file.write(accession + '\t' + live_annotations[accession]["dbname"] + '\t unknown \n')
+                print(live_annotations[accession])
+def main():
+    """
+    Main function to handle command-line arguments and output the result.
+    """
+    parser = argparse.ArgumentParser(description='Fetch assembly accessions from NCBI BioProject and report the number of corresponding annotations in rapid.ensembl.org.')
+    parser.add_argument('bioproject_id', type=str, help='NCBI BioProject ID')
+    parser.add_argument('--haploid', action='store_true', help='Fetch only haploid assemblies')
+    parser.add_argument('--report_file', type=str, help='Where to write report to', default='./report_file.csv')
+    parser.add_argument('--rank', type=str, help='Taxonomic rank to classify', default='order')
+    
+    args = parser.parse_args()
+
+    bioproject_accessions_taxon = get_assembly_accessions(args.bioproject_id, args.haploid)
+    live_annotations = get_ensembl_live(bioproject_accessions_taxon)
+    live_annotations_classified = get_taxonomy_info(live_annotations, bioproject_accessions_taxon, args.rank)
+    unique_taxon_ids = {details['taxon_id'] for details in live_annotations.values()}
+    print(f"Found {len(bioproject_accessions_taxon)} assemblies under BioProject ID {args.bioproject_id}")
```

Comment:
not sure how we will call this script but if we ant to check a list of bio project ids it may be better having a logging file where we store these messages

Reply:
I can't think of anywhere we might call this script (except for creating the list of dbs needed as input for projects pages - but that will require a bit of re-factoring and an option to 'list_dbs_only' or something like that). I considered adding logging, but the usage of this script means that it makes sense to just print a summary

Reply:
No that's fine in this way in case we need to check more ids we could just call this script in a loop